### PR TITLE
fix(codepipeline): default cross-region S3 buckets allow public access

### LINKED ISSUE #16411: (pipelines): cross region S3 bucket - block public access
The cross region S3 bucket that is created should have block public access enabled by default.

Looking at the synthed template, the block public access is not enabled.

### Use Case

Ensure s3 block public access protection at the bucket level.

### Proposed Solution

Include for any cross region s3 buckets.

        "PublicAccessBlockConfiguration": {
          "BlockPublicAcls": true,
          "BlockPublicPolicy": true,
          "IgnorePublicAcls": true,
          "RestrictPublicBuckets": true
        }

### Other

* [ ] :wave: I may be able to implement this feature request
* [ ] :warning: This feature might incur a breaking change

---

This is a :rocket: Feature Request

### PR DESCRIPTION
**Pull request by david-richer-adsk**
The cross region S3 buckets that are created should have block public access by default.

Fixes #16411

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by david-richer-adsk** [comment]
Same approach as for the artifact bucket: https://github.com/aws/aws-cdk/blob/master/packages/@aws-cdk/aws-codepipeline/lib/pipeline.ts#L382

File: packages/@aws-cdk/aws-codepipeline/lib/private/cross-region-support-stack.ts [hunk 1]
```diff
@@ -77,6 +77,7 @@ export class CrossRegionSupportConstruct extends Construct {
       bucketName: cdk.PhysicalName.GENERATE_IF_NEEDED,
       encryption: encryptionAlias ? s3.BucketEncryption.KMS : s3.BucketEncryption.KMS_MANAGED,
       encryptionKey: encryptionAlias,
+      blockPublicAccess: new s3.BlockPublicAccess(s3.BlockPublicAccess.BLOCK_ALL),
```
Comment (skinny85) [inline packages/@aws-cdk/aws-codepipeline/lib/private/cross-region-support-stack.ts:80]: Pretty sure this should be:

```suggestion
      blockPublicAccess: s3.BlockPublicAccess.BLOCK_ALL,
```

**Review by skinny85 — CHANGES_REQUESTED** [review]
Thanks for the contribution @david-richer-adsk!

**Review by skinny85 — APPROVED** [review]

**Review by skinny85 — CHANGES_REQUESTED** [review]
We also need a unit test for this one 🙂.

**Comment by david-richer-adsk** [comment]
> We also need a unit test for this one 🙂.

Does this look like the right place to add a check? https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/aws-codepipeline/test/cross-env.test.ts#L132

**Comment by skinny85** [comment]
> > We also need a unit test for this one 🙂.
> 
> Does this look like the right place to add a check? https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/aws-codepipeline/test/cross-env.test.ts#L132

Yes, that's probably a good spot. You can use the `toHaveResourceLike()` helper, like [here](https://github.com/aws/aws-cdk/blob/b606a2321769d5e8f15072a62848aaba35bb1d35/packages/%40aws-cdk/aws-codepipeline/test/cross-env.test.ts#L196-L198).

**Comment by david-richer-adsk** [comment]
Should be good  to go, let me know if anything is missing.

File: packages/@aws-cdk/aws-codepipeline/test/cross-env.test.ts [hunk 2]
```diff
@@ -119,7 +119,9 @@ describe.each(['legacy', 'modern'])('with %s synthesis', (synthesisStyle: string
 
         // THEN
         expect(supportStack).not.toHaveResource('AWS::KMS::Key');
-        expect(supportStack).toHaveResource('AWS::S3::Bucket');
+        expect(supportStack).toHaveResourceLike('AWS::S3::Bucket', {
+          PublicAccessBlockConfiguration: {},
```
Comment (skinny85) [inline packages/@aws-cdk/aws-codepipeline/test/cross-env.test.ts:123]: Hmm, not sure I understand this assertion. Shouldn't this be:

```suggestion
          PublicAccessBlockConfiguration: {
            BlockPublicAcls: true,
            BlockPublicPolicy: true,
            IgnorePublicAcls: true,
            RestrictPublicBuckets: true,
          },
```

?

**Review by skinny85 — DISMISSED** [review]
Looks good @david-richer-adsk! One question.

File: packages/@aws-cdk/aws-codepipeline/test/cross-env.test.ts (hunk 2 above)
Comment (david-richer-adsk) [reply packages/@aws-cdk/aws-codepipeline/test/cross-env.test.ts:123]: I was just looking for the block `PublicAccessBlockConfiguration ` presence. I will do your changes to also inspect the details of the block

**Comment by david-richer-adsk** [comment]
@skinny85 ready to be merged?

**Review by skinny85 — APPROVED** [review]
Thanks for the contribution @david-richer-adsk!

**Comment by skinny85** [comment]
> @skinny85 ready to be merged?

Yep! In the future, make sure to re-request my review after you're done pushing your changes (there's a button in the top-right corner of the PR page, next to my avatar), this way I won't miss it 🙂.

Thanks,
Adam

**Pull request merged**